src/consensus/pbft/libbyz/replica.cpp

```diff
@@ -479,6 +479,57 @@ void Replica::playback_request(ccf::Store::Tx& tx)
 
   exec_command(vec_exec_cmds, playback_byz_info, 1);
   did_exec_gov_req = did_exec_gov_req || playback_byz_info.did_exec_gov_req;
+  brt.add_request(req.release());
+}
+
+void Replica::populate_certificates(Pre_prepare* pp, bool add_mine)
+{
+  if (pp->seqno() <= 0)
+  {
+    // first pre prepare will not contain proofs for a previous pre prepare
+    return;
+  }
+
+  auto principals_with_proofs = pp->get_prev_pp_valid_principals();
```

Comment:
you can remove this function call & vector allocation by just combining it with line 506